### DISCUSSION TITLE
2000w数据如何实现秒级检索

### DISCUSSION THREAD

**Discussion by bigmouse0001** [AlistGo/alist#4088] [Q&A]
### Please make sure of the following things

- [X] I have read the [documentation](https://alist.nn.ci).
- [X] I'm sure there are no duplicate issues or discussions.
- [X] I'm sure this feature is not implemented.
- [X] I'm sure it's a reasonable and popular requirement.

### Description of the feature / 需求描述

alist有2000w索引数据，每次索引都需要几分钟，这实在是太长了。于是我试着为db文件建立全文索引，但最终失败。

### Suggested solution / 实现思路

1.  对name列建立全文索引，语法错误失败:
creat fulltext index x_search_nodes on x_search_nodes(name)

2.  试图将引擎修改为innodb，再建立全文索引，语法错误失败
alter table x_search_nodes engine=innodb

3. 对name列建立普通索引，查询时间缩短为50s，但仍然过长。
creat index idx_name on x_search_nodes(name)

**Comment by welcome** [comment]
Thanks for opening your first issue here! Be sure to follow the issue template!
